compile.sh

```diff
@@ -31,15 +31,21 @@ else
 	exit 255
 fi
 
-# copy default config from the template
-[[ ! -f "${SRC}"/config-default.conf ]] && cp "${SRC}"/config/templates/config-example.conf "${SRC}"/config-default.conf
-
 # source build configuration file
 if [[ -n $1 && -f "${SRC}/config-$1.conf" ]]; then
 	display_alert "Using config file" "config-$1.conf" "info"
 	# shellcheck source=/dev/null
 	source "${SRC}/config-$1.conf"
 else
+	# copy default config from the template
+	if [[ ! -f "${SRC}"/config-default.conf ]]; then
+		display_alert "Create example config file using template" "config-default.conf" "info"
+		if [[ ! -f "${SRC}"/config-example.conf ]]; then
+			cp "${SRC}"/config/templates/config-example.conf "${SRC}"/config-example.conf || exit 1
+		fi
+		ln -s config-example.conf "${SRC}"/config-default.conf || exit 1
```

Comment:
Don't understand the need for this link?

Reply:
If you have 4 config for your projects, you can easily switch between projects just editing symbolic link.
And you can just use ./compile.sh or ./compile.sh KERNEL_ONLY="no"
IMO it is useful.
I talked to two colleagues before the commit and they also think my proposal is functional.
You decide. If you do not like it, you will not accept it.

Reply:
OK, make sense.